src/WasmDis.ts

```diff
@@ -1171,3 +1237,184 @@ export class NameSectionReader {
     return new NameSectionNameResolver(functionNames, this._functionLocalNames);
   }
 }
+
+export class DevToolsNameGenerator {
+  private _done: boolean;
+  private _hasNames: boolean;
+  private _functionImportsCount: number;
+  private _memoryImportsCount: number;
+  private _tableImportsCount: number;
+  private _globalImportsCount: number;
+
+  private _functionNames: string[][];
```

Comment:
The 2D array approach seems quite heavy memory wise. Given that we likely need to deal with gigantic Wasm modules, I think it'd be good to only record one name per entity.

Reply:
Thanks for the review! I've double checked the section order, and the relevant sections are ordered such that we can only use a 1D array.